services/diversity/src/persistence/richness_nos.js

```diff
@@ -0,0 +1,314 @@
+const { areaTypeKeys } = require('../util/appropriate_keys');
+
+module.exports = (
+  db,
+  {
+    richnessNos,
+  },
+  logger,
+) => ({
+  /**
+   * Find the values for the total number of species in the given area
+   *
+   * @param {String} areaType area type
+   * @param {String | Number} areaId area id
+   *
+   * @returns {Object[]} Number of total inferred and observed species.
+   */
+  findTotalNumberOfSpecies: async (areaType, areaId) => {
+    try {
+      return db('richness_nos as rn')
+        .select(
+          db.raw('coalesce(rn.total_inf, 0) as inferred'),
+          db.raw('coalesce(rn.total_obs, 0) as observed'),
+          db.raw('coalesce(rnr.total_obs, 0) as region_observed'),
+          db.raw('coalesce(rnr.total_inf, 0) as region_inferred'),
+        )
+        .leftJoin('richness_nos_regions as rnr', 'rn.id_region', 'rnr.id_region')
+        .where({ 'rn.geofence_type': areaTypeKeys(areaType), 'rn.geofence_id': areaId });
+    } catch (e) {
+      logger.error(e.stack || e.Error || e.message || e);
+      throw new Error('Error getting data');
+    }
+  },
+
+  /**
+   * Get the thresholds for the total number of species in a given area
+   *
+   * @param {String} areaType area type.
+   * @param {String | Number} areaId area id
+   *
+   * @returns {Object[]} Number of inferred and observed species for the desired group.
+   */
+  findThresholdsTotalNumberOfSpecies: async (areaType, areaId) => {
+    try {
+      const region = db('richness_nos as rn').select('id_region').where({ geofence_id: areaId });
```

Comment:
Don't do the `db('richness_nos as rn')` use the model.

Also why did you choose to make 2 queries instead of a join?

Reply:
I could not get the innerJoin to work with a field alias instead of the name of another or the same table. This is the example I'd tried https://knexjs.org/#Builder-innerJoin.

Reply:
ok, I saw there is a way to make the join with the subquery, but this way is more readable